quickwit-indexing/src/models/raw_doc_batch.rs

```diff
@@ -19,12 +19,27 @@
 
 use std::fmt;
 
-use quickwit_metastore::checkpoint::CheckpointDelta;
+use quickwit_metastore::checkpoint::SourceCheckpointDelta;
 
-#[derive(Default, Clone)]
+#[derive(Clone)]
 pub struct RawDocBatch {
     pub docs: Vec<String>,
-    pub checkpoint_delta: CheckpointDelta,
+    pub checkpoint_delta: SourceCheckpointDelta,
+}
+
+impl RawDocBatch {
+    pub fn new(docs: Vec<String>, checkpoint_delta: SourceCheckpointDelta) -> Self {
+        RawDocBatch {
+            docs,
+            checkpoint_delta,
+        }
+    }
+}
+
+impl Default for RawDocBatch {
```

Comment:
Can't we derive `Default` for `RawDocBatch` if `Vec` and `SourceCheckpointDelta` do?